### PR TITLE
Added create CRUD for research group on admin panel

### PR DESCRIPTION
**Pull request by AlexGit123**

### CODE REVIEW

File: app/Http/Controllers/AdminController.php [hunk 1]
```diff
@@ -15,7 +16,10 @@ class AdminController extends Controller
     public function index()
     {
         $courses = Course::latest()->get();
-        return view('admin', compact('courses'));
+
```
Comment (Koreyoshi159) [inline app/Http/Controllers/AdminController.php:19]: White space

File: app/Http/Controllers/ResearchGroupController.php [hunk 2]
```diff
@@ -20,11 +20,11 @@ public function index()
     /**
      * Show the form for creating a new resource.
      *
-     * @return \Illuminate\Http\Response
+     * @return \Illuminate\Contracts\Foundation\Application|\Illuminate\Contracts\View\Factory|\Illuminate\Contracts\View\View|\Illuminate\Http\Response
```
Comment (Koreyoshi159) [inline app/Http/Controllers/ResearchGroupController.php]: ???

File: app/Http/Controllers/ResearchGroupController.php [hunk 3]
```diff
@@ -33,9 +33,11 @@ public function create()
      * @param  \Illuminate\Http\Request  $request
      * @return \Illuminate\Http\Response
      */
-    public function store(Request $request)
+    public function store(ResearchGroup $researchGroup, Request $request)
```
Comment (Koreyoshi159) [inline app/Http/Controllers/ResearchGroupController.php]: why is there a researchGroup variable

File: resources/views/researchGroup/create.blade.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                <div class="field-body">
+                    <div class="field ">
+                        <div class="control">
+                            <input class="input {{$errors->has('name') ? 'is-danger' : ''}}" type="text" name="name"
+                                   id="name" value="{{old('name')}}">
+
+                            @error('name')
+                            <p class="help is-danger">{{$errors->first('name')}}</p>
+                            @enderror
+
```
Comment (Koreyoshi159) [inline resources/views/researchGroup/create.blade.php]: white space

File: resources/views/admin.blade.php [hunk 5]
```diff
@@ -1,7 +1,6 @@
 @extends('common.master')
 
 @section('content')
-
     <div class="" style="padding: 10%">
```
Comment (Koreyoshi159) [inline resources/views/admin.blade.php]: Inline style?

File: resources/views/admin.blade.php [hunk 6]
```diff
@@ -47,34 +43,68 @@
...
-                                </h1>
-                            </header>
-                            <div class="container">
-                                <table>
-                                    <tbody>
-                                    @foreach($courses as $course)
-                                        <tr>
+            <div class="columns">
+                <div class="column is-6 is-narrow">
+                    <div class="card" style="width: 500px">
```
Comment (awqfrxx) [inline resources/views/admin.blade.php]: inline styling here?
should remove this

File: resources/views/admin.blade.php [hunk 7]
```diff
@@ -28,12 +27,9 @@
                     <div class="card-content">
                         <div class="content">
                             <header class="card-header-title columns">Blogs
-
                                 <h1 style="padding-left: 50%">
```
Comment (Koreyoshi159) [inline resources/views/admin.blade.php]: Inline style

File: resources/views/admin.blade.php [hunk 8]
```diff
@@ -47,34 +43,68 @@
...
-                                    <tbody>
-                                    @foreach($courses as $course)
-                                        <tr>
+            <div class="columns">
+                <div class="column is-6 is-narrow">
+                    <div class="card" style="width: 500px">
+                        <div class="card-content">
+                            <div class="content">
+                                <header class="card-header-title columns">Courses
+                                    <h1 style="padding-left: 50%">
```
Comment (awqfrxx) [inline resources/views/admin.blade.php]: inline styling. remove this

File: routes/web.php [hunk 9]
```diff
@@ -22,8 +24,11 @@
...
 
-Route::get('/admin', [AdminController::class, 'index']);
+Route::get('/admin', [AdminController::class, 'index'])->name('admin');
 
 Route::get('/course/create', [CourseController::class, 'create']);
 Route::post('/course', [CourseController::class, 'store']);
 
+Route::resource('/research_groups', ResearchGroupController::class);
+
+
```
Comment (Koreyoshi159) [inline routes/web.php]: White space?

File: resources/views/admin.blade.php [hunk 10]
```diff
@@ -47,34 +43,68 @@
...
-                                            </td>
-                                            <td>
-                                                <button class="has-text-right"><a href="\">Delete</a></button>
-                                            </td>
-                                        </tr>
-                                    @endforeach
-                                    </tbody>
-                                </table>
+                <div class="column is-6 is-narrow">
+                    <div class="card" style="width: 500px">
```
Comment (awqfrxx) [inline resources/views/admin.blade.php:86]: inline styling again. remove this

**Review by Koreyoshi159 — CHANGES_REQUESTED** [review]
Check out the comments

File: resources/views/admin.blade.php [hunk 11]
```diff
@@ -47,34 +43,68 @@
...
+                        <div class="card-content">
+                            <div class="content">
+                                <header class="card-header-title columns">Research Groups
+                                    <h1 style="padding-left: 50%">
```
Comment (awqfrxx) [inline resources/views/admin.blade.php:90]: inline styling again. remove this

**Review by awqfrxx — CHANGES_REQUESTED** [review]
remove all the inline styling

File: resources/views/researchGroup/create.blade.php [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+            <div class="field is-horizontal">
+                <div class="field-label">
+                    <!-- Left empty for spacing -->
+                </div>
+                <div class="field-body">
+                    <div class="field">
+                        <div class="control">
+                            <button value="Submit" class="button is-info" type="submit">
+                                <p>Create!</p>
```
Comment (avanheeren) [inline resources/views/researchGroup/create.blade.php]: the ! doesn't look very professional, maybe remove that

**Review by avanheeren — CHANGES_REQUESTED** [review]
Resolve feedback

File: app/Http/Controllers/AdminController.php (hunk 1 above)
Comment (Koreyoshi159) [inline app/Http/Controllers/AdminController.php]: Why will there be a white space between 2 variables?

File: app/Http/Controllers/ResearchGroupController.php [hunk 13]
```diff
@@ -82,4 +84,12 @@ public function destroy(ResearchGroup $researchGroup)
     {
         //
     }
+
+    public function validateResearchGroup($request)
+    {
+
```
Comment (Koreyoshi159) [inline app/Http/Controllers/ResearchGroupController.php]: White space

**Review by Koreyoshi159 — APPROVED** [review]
Everything is good we need someone to fix the admin panel's divs xD

**Pull request merged**